src/realm/sync/instruction_replication.cpp

```diff
@@ -83,11 +83,11 @@ Instruction::Payload SyncReplication::as_payload(Mixed value)
         }
     }
     if (type == type_Dictionary) {
```

Comment:
We might have a `Set` in a `Mixed` also, should we not forbid to sync that too.

Reply:
It seems like I have not implemented support for syncing a Set. Perhaps I should add that now.